app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -16,16 +16,162 @@
 
 package com.google.droidjump;
 
+import static androidx.navigation.Navigation.findNavController;
+import android.annotation.SuppressLint;
 import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.view.MotionEvent;
+import android.view.SurfaceHolder;
 import android.view.SurfaceView;
 
 public class GameView extends SurfaceView implements Runnable {
 
+    private boolean isPlaying;
+    private int screenX;
+    private int screenY;
+    private int screenMargin;
+    private SurfaceHolder surfaceHolder;
+    private Droid droid;
+    private int timePoint;
+    private Thread thread;
+    private int levelTimePoints;
+    private int levelSpeed;
+
     public GameView(Context context) {
         super(context);
+        isPlaying = false;
+    }
+
+    public GameView(Context context, int screenX, int screenY, boolean isPlaying) {
+        super(context);
+        receiveLevelDetails();
+        timePoint = 0;
+        surfaceHolder = getHolder();
+        this.screenX = screenX;
+        this.screenY = screenY;
+        this.isPlaying = isPlaying;
+
+        // Margin in px
+        screenMargin = (int) getResources().getDimension(R.dimen.fab_margin);
+
+        // Create droid
+        droid = new Droid(screenMargin, screenY - screenMargin, getResources());
+        droid.setDroidJumpHeight(getResources());
```

Comment:
Can setting droid jump height be done as a part of constructor "new Droid"? You pass resources there anyway. This way we won't have to set it externally from here.

Reply:
You can remove "setDroidJumpHeight(Resources resources)" from Droid class afterwards.